src/mame/machine/elan_eu3a05commonsys.cpp

```diff
@@ -295,6 +295,9 @@ void elan_eu3a05commonsys_device::generate_custom_interrupt(int level)
 
 READ8_MEMBER(elan_eu3a05commonsys_device::nmi_vector_r)
 {
+	if(machine().side_effects_disabled())
+		return 0x00;
+
```

Comment:
Woudln't it be better to do something like:
```
	if (m_custom_nmi)
	{
		return m_custom_nmi_vector >> (offset*8);
	}
	else if(machine().side_effects_disabled())
	{
		return 0x00;
	}
	else
	{
		fatalerror("NMI without custom vector!");
	}
```
so that you’ll at least see the actual value if it’s enabled?

Reply:
I'm looking for some kind of real enable / disable mechanism.  This is just a quick fix to address a crash Tafoid was reporting.

The 'enabled' just get set once a value is written, prior to that I'm not sure what the behavior should be, this could simply be an additional IRQ level built into the CPU that we're basically abusing the existing NMI to trigger - the enable flag gets set once the vector is written so that no invalid NMIs are taken.